app/services/katello/pulp3/content_view_version/export.rb

```diff
@@ -42,15 +65,39 @@ def destroy_exporter(exporter_href)
           api.exporter_api.delete(exporter_href)
         end
 
+        def validate_incremental_export!
+          return if @from_content_view_version.blank?
+          from_exporter = Export.new(smart_proxy: @smart_proxy, content_view_version: @from_content_view_version)
+
+          from_exporter_repos = Hash[from_exporter.repositories.map { |repo| [repo.library_instance_id, version_href_to_repository_href(repo.version_href)] }]
+          to_exporter_repos = Hash[repositories.map { |repo| [repo.library_instance_id, version_href_to_repository_href(repo.version_href)] }]
+
+          invalid_repos_exist = (from_exporter_repos.keys & to_exporter_repos.keys).any? do |repo_id|
+            from_exporter_repos[repo_id] != to_exporter_repos[repo_id]
+          end
+
+          return unless invalid_repos_exist
+
+          fail _("The exported Content View Version '%{content_view} %{current}' cannot be incrementally updated from version '%{from}'."\
+                 " Please do a full export." % { content_view: @content_view_version.content_view.name,
+                                                 current: @content_view_version.version,
+                                                 from: @from_content_view_version.version})
```

Comment:
Not sure if @jturel is starting to have an influence on me, but I don't think the early return here serves a good purpose.  To me an early return says "if this condition is true then the rest of the method is irrelevant so nevermind." But in this case there's only one statement left.  I'd prefer
```rb
if invalid_repos_exist
  ...
end
```